src/components/navigation/SearchBar.tsx

```diff
@@ -13,16 +15,22 @@ const styles: { [identifier: string]: CSSProperties } = {
 };
 
 export default function SearchBar(): ReactElement {
-  const { dispatch } = usePromotionsList();
+  const { state: dropdownState } = useDropdown();
+  const { dispatch: promotionsDispatch } = usePromotionsList();
 
   const onSearch = useCallback(
     (searchQuery: string) => {
-      dispatch({
+      // Reset filters: user cannot be filtering and using a search query simultaneously
+      promotionsDispatch({ type: DispatchAction.RESET_FILTERS });
+      dropdownState.resetCallbacks.forEach((resetDropdown) => resetDropdown());
+
+      // Send search query
+      promotionsDispatch({
         type: DispatchAction.SEARCH_QUERY,
         payload: { searchQuery },
       });
     },
-    [dispatch]
+    [dropdownState.resetCallbacks, promotionsDispatch]
```

Comment:
What does it mean to add `dropdownState.resetCallbacks` to this parameter?

Reply:
Oh yeah this dependency array basically lets us tell the `useCallback` hook to update whenever any one of these variables get updated/changed. In this case, if `dropdownState.resetCallbacks` changes (e.g. another callback gets added) then the previously memorized version of this `onSearch` function would be replaced by a new one with the new `dropdownState.resetCallbacks` reference.

You usually don't have to think too much about what goes in here cause React tells you in the console if any dependencies it expects are missing!

Reply:
Ohhh that's so cool I see thanks for explaining

Reply:
No problem!